src/tlo/methods/healthsystem.py

```diff
@@ -2930,6 +2972,72 @@ def apply(self, population):
+    given an extra budget. This is done for daily capabilities, as a year consists of 365.25 equal days."""
+    def __init__(self, module):
+        super().__init__(module)
+
+    def apply(self, population):
+
+        # get minute salary for the four cadres
+        minute_salary_by_officer_facility_id = self.module.parameters['minute_salary']
+
+        # get current daily minutes and format it to be consistent with minute salary
+        daily_minutes = pd.DataFrame(self.module._daily_capabilities).reset_index().rename(
+            columns={'index': 'facilityid_officer'})
+        daily_minutes[['Facility_ID', 'Officer_Type_Code']] = daily_minutes.facilityid_officer.str.split(
+            pat='_', n=3, expand=True)[[1, 3]]
+        daily_minutes['Facility_ID'] = daily_minutes['Facility_ID'].astype(int)
+
+        # get daily cost per officer per facility id
+        daily_cost = minute_salary_by_officer_facility_id.merge(
+            daily_minutes, on=['Facility_ID', 'Officer_Type_Code'], how='outer')
+        daily_cost['Total_Cost_Per_Day'] = daily_cost['Minute_Salary_USD'] * daily_cost['Total_Minutes_Per_Day']
+
+        # get daily cost per officer type of the four cadres
+        daily_cost = daily_cost.groupby('Officer_Type_Code').agg({'Total_Cost_Per_Day': 'sum'}).reset_index()
+        daily_cost = daily_cost.loc[daily_cost.Officer_Type_Code.isin(
+            ['Clinical', 'DCSA', 'Nursing_and_Midwifery', 'Pharmacy'])]
+
+        # get total daily cost of each of the four cadres
+        total_cost_clinical = daily_cost.loc[daily_cost.Officer_Type_Code == 'Clinical', 'Total_Cost_Per_Day'].sum()
+        total_cost_dcsa = daily_cost.loc[daily_cost.Officer_Type_Code == 'DCSA', 'Total_Cost_Per_Day'].sum()
+        total_cost_nursing = daily_cost.loc[
+            daily_cost.Officer_Type_Code == 'Nursing_and_Midwifery', 'Total_Cost_Per_Day'].sum()
+        total_cost_pharmacy = daily_cost.loc[daily_cost.Officer_Type_Code == 'Pharmacy', 'Total_Cost_Per_Day'].sum()
+
+        # get daily extra budget for this year = 4.2% * total cost
+        daily_extra_budget = 0.042*(total_cost_clinical + total_cost_dcsa + total_cost_nursing + total_cost_pharmacy)
+
+        # get proportional daily extra budget for each of the four cadres
+        daily_extra_budget_by_officer = [
+            daily_extra_budget * i for i in self.module.parameters['HR_expansion_by_officer_type']]
+
+        # get the scale up factor for each cadre, assumed to be the same for each facility id of that cadre
+        sf_clinical = (total_cost_clinical + daily_extra_budget_by_officer[0])/total_cost_clinical
+        sf_dcsa = (total_cost_dcsa + daily_extra_budget_by_officer[1]) / total_cost_dcsa
+        sf_nursing = (total_cost_nursing + daily_extra_budget_by_officer[2]) / total_cost_nursing
+        sf_pharmacy = (total_cost_pharmacy + daily_extra_budget_by_officer[3]) / total_cost_pharmacy
```

Comment:
Might create a logger for these scale up factors for years 2020-2030; the internally calculated scale up factors could be compared with those calculate externally (will write an individual script for this calculation).

Reply:
Logger created. Below is the comparison of scale up factors from healthsystem logger and from manual calculation, which seems generally consistent. The minor difference should be due to the former is calculated using daily minutes (where total annual minutes are evenly distributed to 365.25 days and scaled down by 'capabilities_coefficient') formatted within healthsystem module and minute-salary, whereas the latter is calculated using total staff counts and annual salary.

![image](https://github.com/user-attachments/assets/f5442747-bfb8-474a-a0ed-56b9b46dd941)

Reply:
The difference of the above two dataframes may be mainly due to extra budget fractions saved and read as two-decimal floats, which are not exactly the same to the fractions. For example, for this scenario S_2 here, the four fractions add up to 0.99, not 1.0.

Reply:
Solution added: instead of save as csv and reload the extra budget fractions data, now directly import the fractions data from the script that calculate them, so that floats do not have lose decimals.

Will check again the differences as above once all scenarios are confirmed and job is re-run.